backend/simple_unified_startup.py

```
@@ -12,14 +13,33 @@
+        try:
+            from run_migrations import MigrationRunner
+            migration_runner = MigrationRunner(database_url)
+            migration_success = await migration_runner.run_migrations()
+            if migration_success:
+                logger.info("✅ Database migrations applied successfully")
+            else:
+                logger.error("❌ Database migrations failed - Test monitor requires all migrations")
+                raise Exception("Migration failure: Test Results Dashboard requires successful migrations for auto_fixable and error_message columns")
+        except ImportError as import_error:
+            logger.error(f"❌ Migration system not available: {import_error}")
+            raise Exception("Migration system required: Cannot start without migration support for database-driven features")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)